native/src/main/java/io/ballerina/persist/cmd/Generate.java

```diff
@@ -0,0 +1,252 @@
+            errStream.println("The current directory is not a Ballerina project!");
+            return;
+        }
+        try {
+            ArrayList<Entity> entityArray = readBalFiles();
+            if (entityArray.size() == 0) {
+                errStream.println("Current directory doesn't contain any persist entities");
+            } else {
+                for (Entity entity : entityArray) {
+                    entity.setPackageName(balProject.currentPackage().descriptor().org().value() + "/"
+                            + balProject.currentPackage().descriptor().name().value());
+                    generateClientBalFile(entity);
+                }
+                generateConfigurationBalFile();
+            }
+        } catch (Exception e) {
+            errStream.println(e.getMessage());
+        }
+    }
+
+    private ArrayList<Entity> readBalFiles() throws BalException {
+        ArrayList<Entity> returnMetaData = new ArrayList<>();
+        Path dirPath = Paths.get(this.sourcePath);
+        List<Path> fileList;
+
+
+        try (Stream<Path> walk = Files.walk(dirPath)) {
+            if (walk != null) {
+                fileList = walk.filter(Files::isRegularFile).collect(Collectors.toList());
+                if (hasSyntacticDiagnostics(Paths.get(this.sourcePath)) ||
+                        hasSemanticDiagnostics(Paths.get(this.sourcePath), this.projectEnvironmentBuilder)) {
+                    throw new BalException("Error occurred while building the project!");
+                }
+                for (Path filePath : fileList) {
+                    if (filePath.toString().endsWith(".bal")) {
+                        String[] pathElements = filePath.toString().strip().split(Pattern.quote(File.separator));
+                        String module = "";
+                        String[] dirElements = this.sourcePath.split(Pattern.quote(File.separator));
+                        if (pathElements.length < 2) {
+                            module = "";
+                        } else if (!Arrays.asList(dirElements).contains(pathElements[pathElements.length - 2])) {
+                            module = pathElements[pathElements.length - 2];
+                        }
+                        ArrayList<Entity> retData = BalSyntaxTreeGenerator.readBalFiles(filePath, module);
+                        if (retData.size() != 0) {
+                            returnMetaData.addAll(retData);
+                        }
+                    }
+                }
+                return returnMetaData;
+            }
+        } catch (IOException e) {
+            throw new BalException("Error occurred while reading bal files!");
+        }
+        return new ArrayList<>();
+    }
+
+    private void generateClientBalFile(Entity entity) throws BalException {
+        SyntaxTree balTree = BalSyntaxTreeGenerator.generateBalFile(entity);
+        String clientPath;
+        if (entity.getModule().equals("")) {
```

Comment:
Shall we make the `module` an optional field? like `Optional<String> module`. then we can check whether it is present easily

Reply:
Updated in [a7ec806](https://github.com/ballerina-platform/persist-tools/pull/15/commits/a7ec80640d0620dc8cc6c925a90af40a3296d374)